packages/website/blog/2022-12-05-asts-and-typescript-eslint.md

```diff
@@ -0,0 +1,124 @@
+
+- TS's AST is optimized for its use case of parsing incomplete code and typechecking.
+- ESTree is unoptimized and intended for "general purpose" use-cases of traversing the AST.
+
+ESLint rules are by default only given nodes in the ESTree AST format - which has no knowledge of TypeScript-specific syntax such as interfaces.
+On the other hand, TypeScript's type checking APIs require nodes in the TypeScript AST format.
+
+### Enter TSESTree
+
+To resolve the incompatibilities between ESTrees and the TypeScript AST typescript-eslint provides its own [`@typescript-eslint/parser` package](https://typescript-eslint.io/architecture/Parser.mdx) which:
+
+1. First parses TypeScript syntax into a TypeScript AST
+1. Creates an ESTree AST based on that TypeScript AST
+1. Keeps track of equivalent nodes across each AST
+
+By creating both an ESTree AST and a TypeScript AST, the typescript-eslint parser allows ESLint rules to work with TypeScript code.
+That's why the [Getting Started guide](https://typescript-eslint.io/getting-started) for typescript-eslint has you specify `parser: '@typescript-eslint/parser'` in your ESLint config!
+
+We commonly refer to the ESTree format that also includes TypeScript-specific syntax as **TSESTree**.
+
+### AST Playground
+
+The [TypeScript ESLint playground](https://typescript-eslint.io/play#showAST=es) contains an AST explorer that generates an interactive AST for any code entered into the playground.
+You can activate it under _Options_ > _AST Explorer_ on its left sidebar by selecting the value of _AST Viewer_.
+
+## Further Resources
+
+You can play more with various other ASTs on [astexplorer.net], including those for other languages such as CSS and HTML.
+
+The [AST Wikipedia article](https://en.wikipedia.org/wiki/Abstract_syntax_tree) has a great deal more context and history on ASTs.
+
+### Glossary
+
+Putting together all the terms introduces in this article:
+
+- **AST (Abstract Syntax Tree)**: An object representation of your code's syntax.
+- **Espree**: ESLint's built-in parser that outputs an ESTree-shaped AST.
+- **ESTree**: The AST specification used by ESLint and other common JavaScript tools.
+- **Node Type**: What kind of code syntax an AST node refers to, such as _BinaryExpression_ or _Literal_.
+- **Node**: A single range of code syntax in an AST.
+- **Parser**: A tool that reads in a string and outputs an AST.
+- **TSESTree**: Our extension to the ESTree AST format that also includes TypeScript-specific syntax.
+
+### TypeScript Lint Rules and ASTs
+
+Interested in how these ASTs work with ESLint rules?
+We collaborated with our friends at Sourcegraph on a [Tour de Source on TypeScript ESLint](https://sourcegraph.com/notebooks/Tm90ZWJvb2s6MTA2OA==).
+Read on to learn how ESLint rules use ASTs to analyze code files and, thanks to `@typescript-eslint/parser`, call TypeScript's type checking APIs to analyze code.
```

Comment:
It's been bugging me that the article kinda fizzled at the end... this adds in an actual conclusion.